### DISCUSSION TITLE
How do I schedule a workflow 20 minutes from now?

### DISCUSSION THREAD

**Discussion by emtoner** [community/community#34392] [Actions]
Hey folks, is anyone aware of any way in Github Actions to wait ~20 minutes and then run a workflow? I'm not talking about a cron where it goes off every day at X time.

We have to:
1. Run a workflow to build & upload a package to the App Store
2. Wait roughly 20 minutes for it to be processed.
3. Run a workflow to modify the resulting package.

Due to resource constraints my runner can't be unavailable for that 20 minutes, I need a way to release the existing workflow and kick the rest off to a different workflow 20 minutes from now. Detaching from the parent PID or scheduling something on the runner to kick off a separate workflow might work, but I lose Github feedback if something goes wrong.

**Comment by spenserblack** [comment]
This action has some documented suggestions for making a workflow wait: https://github.com/jakejarvis/wait-action#%EF%B8%8F-this-action-is-unmaintained

**Reply by emtoner** [reply]
Ah thanks, but I can't hold up our runner for this amount of time that's why I'm trying to detach the process.

Basically I'm trying to do something like this:
(sleep 1800 && trigger_workflow.sh) &

Given there is some form of Workflow scheduling via scheduled cronjobs, I thought it could be possible there's another form of one off scheduling I could do via the Github API or something to that effect.

**Reply by spenserblack** [reply]
Got it, I should have read your requirements more carefully. As far as I know a built-in way to delay a run for a certain amount of time after it's triggered doesn't exist. The only other way that I know of is, as you mentioned, using a detached process to [start a separate workflow](https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event).

This sounds like a reasonable feature request, though.
